webserver/webserver/src/main/java/io/helidon/webserver/BareResponseImpl.java

```diff
@@ -293,26 +263,24 @@ private void completeInternalPipe(boolean wasClosed, Throwable throwable) {
             completeResponseFuture(throwable);
             return;
         }
-        requestEntityAnalyzed = requestEntityAnalyzed.thenApply(listener -> {
-            requestContext.runInScope(() -> {
-                if (ChannelFutureListener.CLOSE.equals(listener)) {
-                    if (LOGGER.isLoggable(Level.FINEST)) {
-                        LOGGER.finest(log("Closing with an empty buffer; keep-alive: false", channel));
-                    }
-                } else {
-                    LOGGER.finest(() -> log("Writing an empty last http content; keep-alive: true"));
-                    channel.read();
-                }
-                writeLastContent(throwable, listener);
-            });
-            return listener;
-        });
+        if (keepAlive) {
+            log(Level.FINEST, "Writing an empty last http content; keep-alive: true");
+
+            writeLastContent(throwable, ChannelFutureListener.CLOSE_ON_FAILURE);
```

Comment:
If closing connection which has been asked by client to be kept alive, this would not close it.